File: projects/03. Software_Technologies - july 2018/13. CSharp-ASP.NET-MVC-Overview-Calculator/CalculatorApp/Controllers/HomeController.cs
using Microsoft.AspNetCore.Mvc;
using CalculatorApp.Models;
using CalculatorApp.Models.ViewModels;

namespace CalculatorApp.Controllers
{
    public class HomeController : Controller
    {
        public IActionResult Index(Calculator calculator)
        {
            return View(calculator);
        }

        [HttpPost]
        public IActionResult Calculate(CalculatorViewModel viewModel)
        {
            Calculator calculator = new Calculator
            {
                LeftOperand = viewModel.LeftOperand,
                RightOperand = viewModel.RightOperand,
                Operator = viewModel.Operator
            };

            calculator.CalculateResult();
            return RedirectToAction("Index", calculator);
        }
    }
}

File: projects/Software_Technologies/Java/Calculator/src/main/java/com/softuni/controller/HomeController.java
package com.softuni.controller;

import com.softuni.entity.calculator;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class HomeController {

    @GetMapping("/")
    public String index(Model model) {
        model.addAttribute("operator", "+");
        model.addAttribute("view", "views/calculatorForm");
        return "base-layout";
    }

    @PostMapping("/")
    public String calculate(@RequestParam String leftOperand,
                            @RequestParam String operator,
                            @RequestParam String rightOperand,
                            Model model) {

        double num1;
        double num2;

        try {
            num1 = Double.parseDouble(leftOperand);
        } catch (NumberFormatException ex) {
            num1 = 0;
        }

        try {
            num2 = Double.parseDouble(rightOperand);
        } catch (NumberFormatException ex) {
            num2 = 0;
        }


        calculator calculator = new calculator(num1, operator, num2);

        double result = calculator.calculateResult();

        model.addAttribute("leftOperand", calculator.getLeftOperand());
        model.addAttribute("operator", calculator.getOperator());
        model.addAttribute("rightOperand", calculator.getRightOperand());

        model.addAttribute("result", result);

        model.addAttribute("view", "views/calculatorForm");

        return "base-layout";
    }
}
